buffer.c

```diff
@@ -3332,6 +3332,10 @@ evbuffer_setcb(struct evbuffer *buffer, evbuffer_cb cb, void *cbarg)
 	if (cb) {
 		struct evbuffer_cb_entry *ent =
 		    evbuffer_add_cb(buffer, NULL, cbarg);
+		if (!ent) {
+			EVBUFFER_UNLOCK(buffer);
+			return;
```

Comment:
We probably want to indicate there was an error somehow - probably by changing the signature of the function to return an error int (I think this would be fully backwards-compatible).

Reply:
> by changing the signature of the function to return an error int

It's more helpful to check errors.I will modify my submission.